symphony/bdk/core/service/message/message_parser.py

```diff
@@ -0,0 +1,104 @@
+"""This module contains a set of util functions that can be used on incoming V4Message such as:
+extracting entities like Mentions, Hashtags, Cashtags, Emojis and extracting text content from presentationML
+contained in the message.
+"""
+import json
+
+from enum import Enum
+from json import JSONDecodeError
+from typing import Dict
+from xml.etree import ElementTree as ET
+from xml.etree.ElementTree import ElementTree, fromstring, tostring
+
+from symphony.bdk.core.service.exception import MessageParserError
+from symphony.bdk.gen.agent_model.v4_message import V4Message
+
+
+def get_text_content_from_message(message: V4Message) -> str:
+    """Get text content from PresentationML on incoming messages
+
+    :param message: message containing the PresentationML to be parsed
+    :return: the message text content extracted from the given PresentationML
+    """
+
+    unparsable_presentationml = "Unable to parse the PresentationML, it is not in the correct format."
```

Comment:
should we move this to the except block?

Reply:
yes, it makes sense since It is used only there. I put it directly inside the raise exception line